zerver/views/message_edit.py

```diff
@@ -203,13 +205,20 @@ def update_message_backend(request: HttpRequest, user_profile: UserMessage,
 
         new_stream = get_stream_by_id(stream_id)
 
-    number_changed = do_update_message(user_profile, message, new_stream,
-                                       topic_name, propagate_mode,
-                                       send_notification_to_old_thread,
-                                       send_notification_to_new_thread,
-                                       content, rendered_content,
-                                       prior_mention_user_ids,
-                                       mention_user_ids, mention_data)
+    try:
+        number_changed = do_update_message(user_profile, message, new_stream,
+                                           topic_name, propagate_mode,
+                                           send_notification_to_old_thread,
+                                           send_notification_to_new_thread,
+                                           content, rendered_content,
+                                           prior_mention_user_ids,
+                                           mention_user_ids, mention_data)
+    except DatabaseError:
+        # DatabaseError("Save with update_fields did not affect any rows.") occurs
+        # when the message is deleted in a race between calling access_message
+        # and saving the message.
+        # We use select_for_update to prevent this, but catch the error nonetheless.
```

Comment:
I don't understand this part, if we're doing SELECT FOR UPDATE to prevent this, why are we catching? Isn't it supposed to be an impossible case? (and if somehow it's possible, then we missed something here and thus do want the exception to propagate to detect the bug)

Reply:
> Isn't it supposed to be an impossible case?
Yes, it will be impossible. Catching shouldn't be required. Should I remove it?
I looked at a few other PRs which made changes (in the frontend) so that exceptions would not be thrown, and they still caught the exceptions. I thought it was a part of good programming practice 😅

Reply:
Interesting, could you link some examples of those commits?

Reply:
#11351 is what I came across.

Reply:
Ah, those weren't really removing the possibility of exceptions occuring, so they did have to handled. I think we should remove this block here - it's not supposed to be possible to get there after all.

Reply:
Ok, makes sense. I'll remove that. Thanks!